### ISSUE TITLE
RFC: object store cache management or track health state of object stores

### ISSUE THREAD

**Issue by bgruening** [galaxyproject/galaxy#14977]
EU has a test user that uses S3, similar to what ORG has with iRODS. We are discussing internally with the IT department how large the cache dir should be. But we have no experience with how much data will be in this cache, nor what is performance. 

Currently, we use tmpwatch, and I understand that we should keep the complexity to monitor storage systems out of Galaxy, if we can easily use solutions like tmpwatch. 
The catch is that we can not predict the size of such cache storage and it's highly dependent on the number/type of jobs and I guess also how many times a user views the data in other ways.

The question I have is if Galaxy should keep track of the health/capacity state of a storage system and react accordingly or can we come up with a smarter external way than tmpwatch?

Afaik the short-term-storage is already cleaning up on its own.

xref: https://github.com/galaxyproject/galaxy/pull/14533

**Comment by mvdbeek** [comment]
One thought is to stage inputs into working directory (but duplicate data transfer possibly), the other is tmpwatch at 10% of capacity or something like that (unlikely to run into problems).

**Comment by martenson** [comment]
At RECETOX I tried to optimize the cache size for swift/s3 for some time while also making the code a bit less broken, but eventually I ended up just finding a big storage and using tmpreaper to keep it within that size. It is certainly possible to do better here and the bigger the instance the more important this problem is. :/

**Comment by martenson** [comment]
@mvdbeek I did when using galaxy-managed cache. I did not when unlimited+tmpreaper, but it is not an interesting data point sadly, because I overshot the cache size by a lot (on purpose).

**Comment by mvdbeek** [comment]
> I did when using galaxy-managed cache

Yeah, that's out of the question, this is just not good design, but I'm glad to hear tmpreaper worked fine. If we want to improve on that (not sure we have to!) I would look at the currently non-terminal jobs and update the cached files' timestamp.

**Comment by martenson** [comment]
> If we want to improve on that

I think we should. The removing of broken logic from galaxy-managed cache seems already underway but giving deployers documentation about this and an example of the script you mentioned would go a long way I think.
